src/Simulation/qdk_sim_rs/Cargo.toml.template

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+[package]
+name = "qdk_sim"
```

Comment:
This is perhaps a bit of a nitpick, but would it make sense to use a name other than `qdk_sim` for the crate? Knowing the project it makes sense, but imagining coming at it fresh I might find it surprising that this crate does not contain all the simulators from the QDK. Are there long term plans to include the fullstate simulator as well?

In some ways, this is handled by the `-alpha` or `-beta` in the semantic versioning, so perhaps we can make a similar exception like what was done for the QIR Runtime package and ensure that this crate keeps the `-alpha` in its version even when published as part of a beta or release build?

On a similar note, I'm less familiar with the naming convention for crates, but maybe it would make sense to name this something like `qdk_sim_experimental` and then when we are ready to publish the non-experimental version we use the `qdk_sim` name along with the updated description and documentation?

Reply:
Don't think it's a nitpick at all, thanks for raising this! Perhaps for now, `qdk_sim_experimental` would be the cleanest crate name (can keep the actual DLL the same name since it gets renamed by the pipeline already). That seems to match crates like `serde_json_experimental`, or all the `wasm` experimental crates, such that I think that'd clearly communicate to users the maturity and support levels of the crate. At that point, forcing `-alpha` even on RC builds may be overkill? Anyway, let me go on and make that change, then. Thanks again for raising this!

Reply:
That approach sounds good!

Reply:
Awesome, let me go on and do that, then. I'll apply the same to the Python interface as well for uniformity. Thanks again for raising this!

Reply:
Addressed in 42c03efb180786feb2d61589a1c46a61206209d3. Thank you!